be/src/vec/functions/array/function_array_set.h

```diff
@@ -0,0 +1,223 @@
+            dst.nullable_col = reinterpret_cast<ColumnNullable*>(array_nested_column.get());
+            dst.nested_nullmap_data = &dst.nullable_col->get_null_map_data();
+            dst.nested_col = dst.nullable_col->get_nested_column_ptr().get();
+        } else {
+            array_nested_column = left_data.nested_col->clone_empty();
+            dst.nested_col = array_nested_column.get();
+        }
+        auto dst_offsets_column = ColumnArray::ColumnOffsets::create();
+        dst.offsets_col = dst_offsets_column.get();
+        dst.offsets_ptr = &dst_offsets_column->get_data();
+
+        ColumnPtr res_column;
+        if (_execute_internal<ColumnString>(dst, left_data, right_data) ||
+            _execute_internal<ColumnDate>(dst, left_data, right_data) ||
+            _execute_internal<ColumnDateTime>(dst, left_data, right_data) ||
+            _execute_internal<ColumnUInt8>(dst, left_data, right_data) ||
+            _execute_internal<ColumnInt8>(dst, left_data, right_data) ||
+            _execute_internal<ColumnInt16>(dst, left_data, right_data) ||
+            _execute_internal<ColumnInt32>(dst, left_data, right_data) ||
+            _execute_internal<ColumnInt64>(dst, left_data, right_data) ||
+            _execute_internal<ColumnInt128>(dst, left_data, right_data) ||
+            _execute_internal<ColumnFloat32>(dst, left_data, right_data) ||
+            _execute_internal<ColumnFloat64>(dst, left_data, right_data) ||
+            _execute_internal<ColumnDecimal128>(dst, left_data, right_data)) {
+            res_column = assemble_column_array(dst);
+            if (res_column) {
+                res_ptr = std::move(res_column);
+                return Status::OK();
+            }
+        }
+        return Status::RuntimeError("Unexpected columns: {}, {}", left_data.nested_col->get_name(),
+                                    right_data.nested_col->get_name());
+    }
+
+private:
+    template <typename ColumnType>
+    static bool _execute_internal(ColumnArrayMutableData& dst, const ColumnArrayExecutionData& left_data,
+                                    const ColumnArrayExecutionData& right_data) {
+        using Impl = OpenSetImpl<operation, ColumnType>;
+        if (!check_column<ColumnType>(*left_data.nested_col)) {
+            return false;
+        }
+        constexpr auto apply_left_first = Impl::Action::apply_left_first;
+        size_t current = 0;
+        Impl impl;
+        for (size_t row = 0; row < left_data.offsets_ptr->size(); ++row) {
+            size_t count = 0;
+            size_t left_off = (*left_data.offsets_ptr)[row - 1];
+            size_t left_len = (*left_data.offsets_ptr)[row] - left_off;
+            size_t right_off = (*right_data.offsets_ptr)[row - 1];
+            size_t right_len = (*right_data.offsets_ptr)[row] - right_off;
+            if (apply_left_first) {
+                impl.apply(left_data, left_off, left_len, dst, &count, true);
```

Comment:
```suggestion
                impl.apply<true>(left_data, left_off, left_len, dst, &count);
```

maybe it is possible to pass left_or_rigth info through template argument.

Reply:
Good suggestion. I will use this.